File: C++/151_ReverseWordsInAString.cpp
class Solution {
public:
    string reverseWords(string s) {
        char * pch = strtok (&s[0]," ");
        if (pch == NULL) return "";
        std::list<string> list; 
        while (pch != NULL){
           list.push_front(string(pch));
           pch = strtok(NULL, " ");
        }
        string ans = "";
        for (auto word : list) 
            ans += ( word + " "); 
        ans.pop_back();
        return ans;
    }
};

File: C#/151_ReverseWordsInAString.cs
public class Solution {
    public string ReverseWords(string s) {
        s = s.Trim();
        string[] splitted = s.Split(' ');
        StringBuilder ans = new StringBuilder("", s.Length);
        int i = splitted.Length;
        while (i --> 0) {
            if (splitted[i] == "") continue;
            ans.Append(splitted[i]);
            if (i > 0) ans.Append(" ");
        }         
        return ans.ToString();
    }
}
